crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -1473,6 +1476,12 @@ def non_disruptive_upgrade_compute_nodes(names)
               break
             end
           end
+          if nodes_to_upgrade.empty?
+            raise_node_upgrade_error(
+              "None of the nodes selected for upgrade can be upgraded now. " \
+              "Check /var/log/crowbar/production.log for details."
```

Comment:
Hm, although this is true in fact, we should somehow direct user to the node with a failure, don't you think?

Reply:
I really think we could end in this place only by failing the live migration of the first node in the `compute_nodes` list. So we know what happened and should be able to tell more.

Reply:
But there should be a line about live migration already in the log, right?

Reply:
It is, but I think we should mention it in the error message if we know it. Going to the logs is next level.

Reply:
I added another line with hint on the live evacuation.

Reply:
I added name of first node to the message. Please test this if you have some env where this scenario can be triggered.

Reply:
Wouldn't it be simpler to just save the node name at the time we catch the timeout exception and exit the loop?

Reply:
Hmm... adding another variable just to print it in some corner case... not sure which is simpler really...

Reply:
In fact, the "None of the nodes selected for upgrade can be upgraded now." message could be wrong.
Consider 2 nodes selected for upgrade. live evacuation fails for node B but not for node A, never mind, we just upgrade A. `compute_nodes` is just one node now and it goes to next cycle. live evacuation again fails for node B and as there's no other node to upgrade left, we exit with an error (correctly). But we should not say nothing got upgraded, because actually one node was.

Reply:
The "now" was supposed to indicate that this includes changes done between the request and now :o)
I reworded the message. Any better?